src/Util.cc

```diff
@@ -216,5 +216,31 @@ void pinAvailableCore() {
     pinThreadToCore(coreId);
 }
 
+/**
+  * Returns the CPU ID of coreId's hypertwin.  If there is no hypertwin, return -1.
```

Comment:
I'm wondering if the interface to this method is too narrow. Is it possible that there can be more than 2 hypertwins on a core? If we think there's any chance of this happening (in the future, if not now), then it would be better to design an interface that can support it.

Rather than querying for a single hypertwin, I'm wondering if you should take a different approach where you simply map each logical core to a physical core, and use that mapping to guide decisions in the CoreArbiter. Let's discuss?